lib/SPIRV/SPIRVReader.cpp

```diff
@@ -4262,6 +4262,20 @@ bool SPIRVToLLVM::transFPGAFunctionMetadata(SPIRVFunction *BF, Function *F) {
     MetadataVec.push_back(ConstantAsMetadata::get(getUInt32(M, Literals[1])));
     F->setMetadata(kSPIR2MD::LoopFuse, MDNode::get(*Context, MetadataVec));
   }
+  if (BF->hasDecorate(internal::DecorationMathOpDSPModeINTEL)) {
+    std::vector<Metadata *> MDVecDSPPref;
+    auto Literals =
+        BF->getDecorationLiterals(internal::DecorationMathOpDSPModeINTEL);
+    MDVecDSPPref.push_back(ConstantAsMetadata::get(getUInt32(M, Literals[0])));
+    F->setMetadata(kSPIR2MD::PreferDSP, MDNode::get(*Context, MDVecDSPPref));
+
+    if (auto Propagate = Literals[1]) {
```

Comment:
Is it safe to directly access `Literals[0]` `Literals[1]` without checking the number of elements in `Literals` vector?

Reply:
As i see, other methods also using direct access without checking Literals's size, but ok.

Reply:
Please note that it might not worth to do a runtime check, but in that case an `assert` would be good